### ISSUE TITLE
Inventory tracking and combos

### ISSUE THREAD

**Issue by aravindet** [hasgeek/boxoffice#109]
Say there is a 2-day ticket, a day 1 ticket and a day 2 ticket. When either day 1 or day 2 tickets are fully sold out, the 2-day ticket should also be marked as fully sold out.

**Bonus feature:** When the day 1 tickets are 90% sold out, stop selling them but continue to offer the 2-day ticket (as that is more profitable).
### Suggested implementation:

```
class SaleItem(...): # What’s currently called Item, except remove quantity
class InventoryItem(...): # Just quantity, number_sold? (if denormalizing)
class InventoryItemMap(): # Mapping
    quantity # over-rides the inventory item quantity, for satisfying bonus feature above
    multiplier # usually one, but if say a combo contains two t-shirts this will be 2
```

If not denormalizing, the query for retrieving availability information along with display items is very complex.

Edit: replaced `DisplayItem` with `SaleItem`

**Comment by shreyas-satish** [comment]
This is definitely a problem, and a good solution. Another solution to address the problem of tracking inventory, is to change the relationship between `Item` and `ItemCollection` to a many-to-many, where the join model can be named as `ListedItem`. If an `item` needs to be marketed differently based on the item collection it is being listed in, the `ListedItem` object can contain these specific details.

My sense is, this solution will require fewer changes to make to the code base.
My sense is, this will

**Comment by jace** [comment]
Related to #96, which can potentially remove the need for `Item` being present in `ItemCollection`. Also, this will fix #75 which calls for renaming `Item` to something less generic.

**Comment by jace** [comment]
Deprecated by #171, which discusses the same idea and includes notes on complexities.
